src/integration/test_suites/smoke_test_fc_ratchet_seed_write_read/smoke_test_fc_ratchet_seed_write_read.c

```diff
@@ -0,0 +1,404 @@
+        goto epilogue;
+      }
+    }
+
+    for (uint32_t addr = partitions[CPTRA_SS_LOCK_HEK_PROD_2].address; addr < partitions[CPTRA_SS_LOCK_HEK_PROD_2].digest_address; addr += 4) {
+      dai_wr(addr, w_data[0], w_data[1], 32, 0);
+      dai_rd(addr, &r_data[0], &r_data[1], 32, 0);
+      if (r_data[0] != w_data[0] && r_data[1] != w_data[1] ) {
+        VPRINTF(LOW, "ERROR: [CPTRA_SS_LOCK_HEK_PROD_2] read_value, does not match written_value\n");
+        goto epilogue;
+      }
+    }
+
+    for (uint32_t addr = partitions[CPTRA_SS_LOCK_HEK_PROD_3].address; addr < partitions[CPTRA_SS_LOCK_HEK_PROD_3].digest_address; addr += 4) {
+      dai_wr(addr, w_data[0], w_data[1], 32, 0);
+      dai_rd(addr, &r_data[0], &r_data[1], 32, 0);
+      if (r_data[0] != w_data[0] && r_data[1] != w_data[1] ) {
+        VPRINTF(LOW, "ERROR: [CPTRA_SS_LOCK_HEK_PROD_3] read_value, does not match written_value\n");
+        goto epilogue;
+      }
+    }
+
+    for (uint32_t addr = partitions[CPTRA_SS_LOCK_HEK_PROD_4].address; addr < partitions[CPTRA_SS_LOCK_HEK_PROD_4].digest_address; addr += 4) {
+      dai_wr(addr, w_data[0], w_data[1], 32, 0);
+      dai_rd(addr, &r_data[0], &r_data[1], 32, 0);
+      if (r_data[0] != w_data[0] && r_data[1] != w_data[1] ) {
+        VPRINTF(LOW, "ERROR: [CPTRA_SS_LOCK_HEK_PROD_4] read_value, does not match written_value\n");
+        goto epilogue;
+      }
+    }
+
+    for (uint32_t addr = partitions[CPTRA_SS_LOCK_HEK_PROD_5].address; addr < partitions[CPTRA_SS_LOCK_HEK_PROD_5].digest_address; addr += 4) {
+      dai_wr(addr, w_data[0], w_data[1], 32, 0);
+      dai_rd(addr, &r_data[0], &r_data[1], 32, 0);
+      if (r_data[0] != w_data[0] && r_data[1] != w_data[1] ) {
+        VPRINTF(LOW, "ERROR: [CPTRA_SS_LOCK_HEK_PROD_5] read_value, does not match written_value\n");
+        goto epilogue;
+      }
+    }
+
+    for (uint32_t addr = partitions[CPTRA_SS_LOCK_HEK_PROD_6].address; addr < partitions[CPTRA_SS_LOCK_HEK_PROD_6].digest_address; addr += 4) {
+      dai_wr(addr, w_data[0], w_data[1], 32, 0);
+      dai_rd(addr, &r_data[0], &r_data[1], 32, 0);
+      if (r_data[0] != w_data[0] && r_data[1] != w_data[1] ) {
+        VPRINTF(LOW, "ERROR: [CPTRA_SS_LOCK_HEK_PROD_6] read_value, does not match written_value\n");
+        goto epilogue;
+      }
+    }
+
+    for (uint32_t addr = partitions[CPTRA_SS_LOCK_HEK_PROD_7].address; addr < partitions[CPTRA_SS_LOCK_HEK_PROD_7].digest_address; addr += 4) {
+      dai_wr(addr, w_data[0], w_data[1], 32, 0);
+      dai_rd(addr, &r_data[0], &r_data[1], 32, 0);
+      if (r_data[0] != w_data[0] && r_data[1] != w_data[1] ) {
+        VPRINTF(LOW, "ERROR: [CPTRA_SS_LOCK_HEK_PROD_7] read_value, does not match written_value\n");
+        goto epilogue;
+      }
+    }
+
+
+    //=================================
+    // Calculate digests:
+    //=================================
+    dai_wr(partitions[CPTRA_SS_LOCK_HEK_PROD_0].digest_address, 0xFF, 0xFF, 32, 0);
```

Comment:
Digests have a granularity of 64 bit:
```suggestion
    dai_wr(partitions[CPTRA_SS_LOCK_HEK_PROD_0].digest_address, 0xFF, 0xFF, 64, 0);
```

Reply:
Done